resources/asciidoctor/lib/definition_hover/extension.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'asciidoctor/extensions'
+
+##
+# Extension for adding a definition to key terms in the docs
+# Must be used inline. Does not have unique block formatting
+#
+# Usage
+#
+#   Foo definition:["word", "definition"]
+#   I like to definition:[run,To move at a speed faster than a walk.]
+#
+class DefinitionAdmonition < Asciidoctor::Extensions::Group
+  MACRO_CONF = [
+    [:definition, 'word', 'definition', nil, nil],
+  ].freeze
+  def activate(registry)
+    MACRO_CONF.each do |(name, revisionflag, tag, message, title_class)|
+      inline = ChangeAdmonitionInline.new message, title_class
+      registry.inline_macro inline, name
+    end
+  end
+
+  ##
+  # Inline change admonition.
+  class ChangeAdmonitionInline < Asciidoctor::Extensions::InlineMacroProcessor
+    use_dsl
+    name_positional_attributes :version, :text
+    format :short
+
+    def initialize(message, extra_title_class)
+      super(nil)
+      @message = message
+      @extra_title_class = extra_title_class
+    end
+
+    def process(parent, _target, attrs)
```

Comment:
Based on examples in the [asciidoctor reference](https://asciidoctor-docs.netlify.app/asciidoctor/1.5/api/block-processor/), we should be able to access a `_target.lines` function that returns the content underneath a custom-defined block. I'm not sure why that doesn't work here. It returns an empty array.

Reply:
Resolved with the help of @gtback. Accessing block content only works within a BlockProcessor. See docs [here](https://www.rubydoc.info/github/asciidoctor/asciidoctor/Asciidoctor/Extensions/BlockProcessor), and an example below:

```
class Shout < Asciidoctor::Extensions::Group
  def activate(registry)
    registry.block ShoutBlock.new :shout
  end

  ##
  # Inline change admonition.
  class ShoutBlock < Asciidoctor::Extensions::BlockProcessor
    use_dsl

    # Stores content passed in from asciidoc in `attrs[:x]`
    # name_positional_attributes :input_word, :input_def
    name_positional_attributes 'vol'
    parse_content_as :simple

    # I don't know what this does, but it's necessary for this widget to work
    # format :short

    def process(parent, _target, attrs)
      # Access attributes passed in from raw asciidoc
      input_word = attrs[:input_word]
      message = attrs[:input_def]

      # on_context :paragraph
      # name_positional_attributes 'vol'


      volume = ((attrs.delete 'vol') || 1).to_i
      create_paragraph parent, (_target.lines.map {|l| l.upcase}), attrs

      # Create a new line block
      # Asciidoctor::Block.new(
      #   parent, :admonition, message, type: 'definition', attributes: {
      #     # 'input_word' => input_word,

      #     # These aren't needed. I'll delete them soon enough
      #     # 'word' => @word,
      #     # 'definition' => @definition,
      #   }
      # )
    end
  end
end
```